python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -719,10 +722,27 @@ def inner(fn: Callable[..., Any]) -> AssetsDefinition:
         }
 
         if specs:
-            internal_deps = {spec.asset_key: spec.deps for spec in specs if spec.deps is not None}
+            internal_deps = {
+                spec.asset_key: {dep.asset_key for dep in spec.deps}
+                for spec in specs
+                if spec.deps is not None
+            }
             props_by_asset_key: Mapping[AssetKey, Union[AssetSpec, AssetOut]] = {
                 spec.asset_key: spec for spec in specs
             }
+            # Add PartitionMappings specified via AssetSpec.deps to partition_mappings dictionary. Error on duplicates
+            for spec in specs:
+                for dep in spec.deps:
+                    if dep.partition_mapping is not None:
+                        if partition_mappings.get(dep.asset_key, None):
+                            if partition_mappings[dep.asset_key] != dep.partition_mapping:
+                                raise Exception(
```

Comment:
recommend doing some `continue` action to avoid excessive nesting

```python
if dep.partition_mapping is None:
   continue

partition_mapping = partition_mappings.get(dep.asset_key)
if not partition_mapping:
   continue
```

etc